### PR TITLE
`azurerm_mariadb_server`: Add support for `ssl_minimal_tls_version_enforced`

### LINKED ISSUE #7899: Support for TLS version in azurerm_mariadb_server
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Hi team,

it could be nice, if we can manage TLS version for Azure MariaDB servers, like it's already the case for MySQL and PostGreSQL.

### New or Affected Resource(s)

* azurerm_mariadb_server

### Potential Terraform Configuration

My suggest is just to add **ssl_minimal_tls_version_enforced** like others database servers ressources

```hcl
resource "azurerm_mariadb_server" "example" {
  name                = "example-mariadb-server"
  location            = azurerm_resource_group.example.location
  resource_group_name = azurerm_resource_group.example.name

  administrator_login          = "mariadbadmin"
  administrator_login_password = "H@Sh1CoR3!"

  sku_name   = "B_Gen5_2"
  storage_mb = 5120
  version    = "10.2"

  auto_grow_enabled             = true
  backup_retention_days         = 7
  geo_redundant_backup_enabled  = false
  public_network_access_enabled = false
  ssl_enforcement_enabled       = true
  ssl_minimal_tls_version_enforced  = "TLS1_2"
}
```

### References

I've no idea exactly limitations for TF against ARM ressources, around SDK Go I think, but at least in ARM template I can see that the option seem not yet exist
https://docs.microsoft.com/en-us/azure/templates/microsoft.dbformariadb/servers

Than you and have a nice day
regards
Alex

### PR DESCRIPTION
**Pull request by favoretti**
```
$ TF_ACC=1 go test -v ./internal/services/mariadb -timeout=1000m -run='TestAccMariaDbServer_basicTenTwo'

=== RUN   TestAccMariaDbServer_basicTenTwo
=== PAUSE TestAccMariaDbServer_basicTenTwo
=== CONT  TestAccMariaDbServer_basicTenTwo
--- PASS: TestAccMariaDbServer_basicTenTwo (296.33s)
PASS
ok  	github.com/hashicorp/terraform-provider-azurerm/internal/services/mariadb	297.574s
```

Fixes #7899

### CODE REVIEW

**Comment by favoretti** [comment]
image

File: website/docs/r/mariadb_server.html.markdown [hunk 1]
```diff
@@ -76,6 +77,8 @@ The following arguments are supported:
 
 * `ssl_enforcement_enabled` - (Required) Specifies if SSL should be enforced on connections. Possible values are `true` and `false`.
 
+* `ssl_minimal_tls_version_enforced` - (Optional) The minimum TLS version to support on the sever. Possible values are `TLSEnforcementDisabled`, `TLS1_0`, `TLS1_1`, and `TLS1_2`. Defaults to `TLS1_2`.
```
Comment (stephybun) [inline website/docs/r/mariadb_server.html.markdown:80]:
```suggestion
* `ssl_minimal_tls_version_enforced` - (Optional) The minimum TLS version to support on the server. Possible values are `TLSEnforcementDisabled`, `TLS1_0`, `TLS1_1`, and `TLS1_2`. Defaults to `TLS1_2`.
```

File: internal/services/mariadb/mariadb_server_resource.go [hunk 2]
```diff
@@ -150,6 +150,13 @@ func resourceMariaDbServer() *pluginsdk.Resource {
 				Required: true,
 			},
 
+			"ssl_minimal_tls_version_enforced": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				Default:      string(servers.MinimalTlsVersionEnumTLSOneTwo),
+				ValidateFunc: validation.StringInSlice(servers.PossibleValuesForMinimalTlsVersionEnum(), false),
+			},
```
Comment (stephybun) [inline internal/services/mariadb/mariadb_server_resource.go:158]: Whilst I agree from a security perspective that this should be defaulted to `TLS1_2`, it looks like the API is defaulting this to `TLSEnforcementDisabled`. Perhaps this should be `Computed` for now and default to `TLS1_2` in 4.x?

**Review by stephybun — CHANGES_REQUESTED** [review]
Thanks for this @favoretti, one minor typo in the docs and also a concern regarding the default value for the property. Once those are resolved this should be good to go though!

**Comment by stephybun** [comment]
@favoretti any updates?

**Comment by favoretti** [comment]
Oh, I’m so sorry, I totally forgot about this one. Will pick it up later
today.

**Comment by favoretti** [comment]
Ah, it seems #20782 got merged ahead of slowpokle me :) Closing this one then.

**Pull request closed without merging**